"Include my Changes" checkbox is not getting reset to enabled

MarekB01 opened the pull request:
https://github.com/BroadleafCommerce/QA/issues/4914

Added comments about a web request scope so checkbox will save state for a whole session

Commit:
BroadleafCommerce/QA#4914
- fixes to save "include my changes" checkbox state between requests in sandbox preview ribbon

MarekB01 committed:
BroadleafCommerce/QA-4914_preview_status_changes

Added comments about a web request scope so checkbox will save state for a whole session

oleksiimiroshnyk commented on common/src/main/java/org/broadleafcommerce/common/web/BroadleafRequestProcessor.java:
```diff
@@ -261,6 +264,7 @@ protected void clearBroadleafSessionAttrs(WebRequest request) {
             request.removeAttribute(BroadleafCurrencyResolverImpl.CURRENCY_VAR, WebRequest.SCOPE_SESSION);
             request.removeAttribute(BroadleafTimeZoneResolverImpl.TIMEZONE_VAR, WebRequest.SCOPE_SESSION);
             request.removeAttribute(BroadleafSandBoxResolver.SANDBOX_ID_VAR, WebRequest.SCOPE_SESSION);
+            request.removeAttribute("blIncludeMyChanges", WebRequest.SCOPE_SESSION);
```
let's extract blIncludeMyChanges to "constant" and re-use, similar to BroadleafSandBoxResolver.SANDBOX_ID_VAR

oleksiimiroshnyk commented on common/src/main/java/org/broadleafcommerce/common/web/BroadleafSandBoxResolverImpl.java:
```diff
@@ -110,6 +109,10 @@ public SandBox resolveSandBox(WebRequest request, Site site) {
         if (BLCRequestUtils.isOKtoUseSession(request)) {
             previousSandBoxId = (Long) request.getAttribute(SANDBOX_ID_VAR, WebRequest.SCOPE_SESSION);
+            // set a web request to scope session
+            if (request.getParameter("blIncludeMyChanges") != null) {
```
let's re-use constant that will be extracted

oleksiimiroshnyk commented on common/src/main/java/org/broadleafcommerce/common/web/BroadleafSandBoxResolverImpl.java:
```diff
@@ -139,6 +142,7 @@ public SandBox resolveSandBox(WebRequest request, Site site) {
                 if (BLCRequestUtils.isOKtoUseSession(request)) {
                     request.removeAttribute(SANDBOX_DATE_TIME_VAR, WebRequest.SCOPE_SESSION);
                     request.removeAttribute(SANDBOX_ID_VAR, WebRequest.SCOPE_SESSION);
+                    request.removeAttribute("blIncludeMyChanges", WebRequest.SCOPE_SESSION);
```
here also

MarekB01 committed:
BroadleafCommerce/QA-4914_preview_status_changes

Added comments about a web request scope so checkbox will save state for a whole session. Refactored code.

oleksiimiroshnyk commented on common/src/main/java/org/broadleafcommerce/common/web/BroadleafRequestProcessor.java:
```diff
@@ -66,6 +65,7 @@ public class BroadleafRequestProcessor extends AbstractBroadleafWebRequestProces
     public static final String SITE_DISABLE_SANDBOX_PREVIEW = "site.disable.sandbox.preview";
 
     private static final String SANDBOX_ID_PARAM = "blSandboxId";
+    private static final String INCLUDE_MY_CHANGES_VAR = "blIncludeMyChanges";
```
I think we can delete this and use BroadleafIncludeMyChangesResolver.INCLUDE_MY_CHANGES_VAR

oleksiimiroshnyk commented on common/src/main/java/org/broadleafcommerce/common/web/BroadleafRequestProcessor.java:
```diff
@@ -168,8 +168,8 @@ public void process(WebRequest request) {
                         url.append(queryString);
-                        if(hsr.getParameter("blIncludeMyChanges")!=null){
-                            url.append("&blIncludeMyChanges="+hsr.getParameter("blIncludeMyChanges"));
+                        if(hsr.getParameter(INCLUDE_MY_CHANGES_VAR) != null){
```
reuse BroadleafIncludeMyChangesResolver.INCLUDE_MY_CHANGES_VAR

oleksiimiroshnyk commented on common/src/main/java/org/broadleafcommerce/common/web/BroadleafRequestProcessor.java:
```diff
@@ -168,8 +168,8 @@ public void process(WebRequest request) {
-                        if(hsr.getParameter("blIncludeMyChanges")!=null){
-                            url.append("&blIncludeMyChanges="+hsr.getParameter("blIncludeMyChanges"));
+                        if(hsr.getParameter(INCLUDE_MY_CHANGES_VAR) != null){
+                            url.append("&blIncludeMyChanges="+hsr.getParameter(INCLUDE_MY_CHANGES_VAR));
```
here. instead of &blIncludeMyChanges=
reuse 
"&"+BroadleafIncludeMyChangesResolver.INCLUDE_MY_CHANGES_VAR+"="

oleksiimiroshnyk commented on common/src/main/java/org/broadleafcommerce/common/web/BroadleafSandBoxResolverImpl.java:
```diff
@@ -110,8 +111,8 @@ public SandBox resolveSandBox(WebRequest request, Site site) {
             // set a web request to scope session
-            if (request.getParameter("blIncludeMyChanges") != null) {
-                request.setAttribute("blIncludeMyChanges", request.getParameter("blIncludeMyChanges"), WebRequest.SCOPE_SESSION);
+            if (request.getParameter(INCLUDE_MY_CHANGES_VAR) != null) {
```
reuse BroadleafIncludeMyChangesResolver.INCLUDE_MY_CHANGES_VAR

oleksiimiroshnyk commented on common/src/main/java/org/broadleafcommerce/common/web/BroadleafSandBoxResolverImpl.java:
```diff
@@ -142,7 +143,7 @@ public SandBox resolveSandBox(WebRequest request, Site site) {
                     request.removeAttribute(SANDBOX_DATE_TIME_VAR, WebRequest.SCOPE_SESSION);
                     request.removeAttribute(SANDBOX_ID_VAR, WebRequest.SCOPE_SESSION);
-                    request.removeAttribute("blIncludeMyChanges", WebRequest.SCOPE_SESSION);
+                    request.removeAttribute(INCLUDE_MY_CHANGES_VAR, WebRequest.SCOPE_SESSION);
```
reuse BroadleafIncludeMyChangesResolver

MarekB01 committed:
BroadleafCommerce/QA-4914_preview_status_changes

Added comments about a web request scope so checkbox will save state for a whole session. Refactored code. Reused a constant

riteshadhikari17 commented:
I am seeing an issue where it is saying access denied but if I refresh it, it works. Here is the video

https://user-images.githubusercontent.com/35038090/225434043-924763b5-2737-45e0-bee2-ac3e3d41ef06.mov

MarekB01 commented:
@riteshadhikari17 it acts quiet strange, I just retested it and it works fine for me, I dont get this 403 error, and changes are passing

riteshadhikari17 commented:
I still see the error after erasing and rebuilding DB and M2. This is the exact error
```
org.thymeleaf.exceptions.TemplateProcessingException: Exception evaluating SpringEL expression: "'locale-' + session.blLocale.localeCode" (template: "layout/homepage" - line 9, col 7)
EL1007E: Property or field 'localeCode' cannot be found on null
```

**Resolved**
Domain was set to `test.localhost` so when previewing the url was `test.test.localhost`. Having incorrect URL was causing url to not be resolve properly.
